### ISSUE TITLE
report2: problem generating report from URL

### ISSUE THREAD

**Issue by rainer-tammer** [sni/Thruk#1011]
Hello,
The report from URL does not work for me.
The report only contains the PDF version of the login page with the banner "session has expired".

Thruk Version: 2.34
icinga2 Version: 2.11.3
PNP4nagios: 0.6.26
cookie auth for thruk/PNP4nagis enabled (single sign-on)

1. Go to "Reporting" and create a new report, source from URL

Sample:  URL: http://icinga2-thruk-host.domain/thruk/cgi-bin/extinfo.cgi?type=2&host=xxx.domain&service=SERVICE&backend=xnnnn#pnp_th2/1586337082/1586427082/0

2. Run the report

3. View the report

Expected: The "Service State Information" / "Performance Graph" page as PDF.

Any help is appreciated.

Bye
  Rainer

**Comment by rainer-tammer** [comment]
One additional note: I have tested this on https://demo.thruk.org/demo/thruk/#cgi-bin/reports2.cgi?#my - it's the same behavior as on my internal installation.

**Comment by rainer-tammer** [comment]
Any ideas?
Is this supposed to work?

**Comment by sni** [comment]
Yes, this is supposed to work. I need to try to reproduce that here...

**Comment by rainer-tammer** [comment]
Hello,
I have just updated to 2.34-2. The problem is still present, so there is no dependency to an older build.

Bye
  Rainer

**Comment by sni** [comment]
When i try this on the demo.thruk site, i simply get no pnp image, but the rest of the page looks ok. But thats something else, right? Is the hole page a "session expired" or only the pnp image?

**Comment by rainer-tammer** [comment]
Hello,
Yes, the problem is primarily the session expired. 
The problem was the same on the demo system (in April).

I had used the following URL:  http://icinga2-thruk-host.domain/thruk/cgi-bin/extinfo.cgi?type=2&host=xxx.domain&service=SERVICE&backend=xnnnn#pnp_th2/1586337082/1586427082/0

Do you have a working sample URL?

Bye
  Rainer

**Comment by sni** [comment]
I just tried, and it kind of works on the demo system. Except, it does not show the pnp images, because there is some javascript involved which changes the pnp url.
If possible, do not use the full absolute url, but the relative url starting with `/thruk/...`.

**Comment by rainer-tammer** [comment]
Hello,
This one does work: /thruk/cgi-bin/status.cgi?host=xxx.xxx.xx&style=detail
This one does work, too: /thruk/cgi-bin/extinfo.cgi?type=1&host=xxx.xxx.xx#my/1591697475/1591787475/0

The pnp images are missing, too.

So is does look good with relative URLs (HTML/PDF).

Do you see any possibility to make the images work?

Thanks in advance.

Bye
  Rainer
